phabricatoremails/render/events/__init__.py

```diff
@@ -0,0 +1,29 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""Describes data structures for public Phabricator revision events.
```

Comment:
The event data structures don't seem to have a structural relationship that can be represented effectively with abstract base classes, so I don't think they're the right call.

So, though a developer won't be able to check an event dataclass's parent for documentation, they'll be able to find the docs by check the event dataclass's module - it's still accessible.

Reply:
> The event data structures don't seem to have a structural relationship that can be represented effectively with abstract base classes, so I don't think they're the right call.

I think the awkwardness is coming from the way `parse` methods behave quite differently across these classes. I should note that since this is interfacing with an external API, I understand the challenge that comes with trying to have clean mappings.

Classes like `MetadataEditedReviewer` and `RevisionAbandoned` and others are doing the exact same thing in `parse`, something that could be refactored (i.e. they are taking a dict, and mapping a value or list of values to a field, directly, and their code is more or less repeated). Others are doing different parsing, validation adding/changing values, and raising exceptions, and some don't have a parse method at all. Those differences can probably be surfaced a little more clearly, perhaps in different methods. Though I think this implementation is OK for this version (v1).

> Fortunately, though a developer won't be able to check an event dataclass's parent for documentation, they'll be able to find the docs by check the event dataclass's module.

We can mark this conversation as resolved; the important bit is that documentation is updated.